src/smolagents/models.py

```diff
@@ -279,7 +279,7 @@ def _prepare_completion_kwargs(
             completion_kwargs["grammar"] = grammar
 
         # Handle tools parameter
-        if tools_to_call_from:
+        if tools_to_call_from and len(tools_to_call_from) > 0:
```

Comment:
I think the second condition is redundant.
- If `len(tools_to_call_from) > 0`
- Then `if tools_to_call_from` is True